File: 2020/day1/day1.jl
using Test
using Combinatorics

function part1(aList)
    x = combinations(aList, 2)
    for item in x
        if reduce(+, item) == 2020
            return reduce(*, item)
        end
    end
end

@test part1((1721, 979, 366, 299, 675, 1456)) == 514579

function part2(aList)
    x = combinations(aList, 3)
    for item in x
        if reduce(+, item) == 2020
            return reduce(*, item)
        end
    end
end

@test part2((1721, 979, 366, 299, 675, 1456)) == 241861950

input = parse.(Int, readlines(open("input.txt")))

print("Part1:", part1(input), "\n")

print("Part2:", part2(input), "\n")

File: 2020/day1/day1.clj

(ns day1)

(slurp "input.txt")

(defn part1 [coll]
(for [x coll y coll
      :when (= (+ x y) 2020)
      :while (< y x)]
  (* x y)))

(defn part1-test []
  (let [coll [1721 979 366 299 675 1456]]
  (part1 coll)  )
  )
(println (part1-test))
